src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -56,15 +58,37 @@ Copyright (C) 2016 Xamarin. All rights reserved.
   <Message Text="IntermediateOutputPath: $(IntermediateOutputPath)" />
 </Target>
 
+<Target Name="_GetProjectResourceDirectory" Returns="@(_ProjectReferenceResourceDirectory)">
+  <ItemGroup>
+    <_ProjectReferenceResourceDirectory Include="$(MSBuildProjectDirectory)\$(MonoAndroidResourcePrefix)"
+      StampFile="$(MSBuildProjectFile)"
+      Condition="!Exists('$(OutputPath)$(TargetName).aar')"
+    />
+  </ItemGroup>
+</Target>
+
+<Target Name="_CollectProjectReferenceResources"
+    Condition="'$(AndroidUseDesignerAssembly)' == 'True' And '$(DesignTimeBuild)' == 'True' "
+>
+  <MSBuild
+      Projects="@(ProjectReference)"
+      Targets="Restore;_GetProjectResourceDirectory"
```

Comment:
Two questions here:
1. Does this really need to run `Restore`?
2. What happens if the `@(ProjectReference)` is a plain `net9.0` project or something old like `netstandard2.0` or PCL?